system-test/storage.ts

```diff
@@ -880,18 +880,19 @@ describe('storage', () => {
 
     let bucket: Bucket;
 
-    before(() => {
-      bucket = storage.bucket('storage-library-test-bucket');
+    before(async () => {
+      [bucket] = await storage.createBucket(generateName());
```

Comment:
This looks like it's creating buckets, but possibly not cleaning them up after.  Do we have code that will come back around and delete it, or is it possible we're leaking buckets (which is a funny turn of phrase now that I think about it)

Reply:
Good catch! I will make that fix.

Reply:
Yep, the after hook deletes all buckets that were created (as long as the "generateName()" function was used for a name).

Reply:
@JustinBeckwith best practice question: should the `after()` clause that deletes the bucket wait for the bucket to actually be deleted? (`return await bucket.delete();` instead of `return bucket.delete()`)

Reply:
So this is important!  I always `await`, and then `return`.  I personally prefer to keep those on separate lines for complete clarity.  This is a good read on the subject:
https://jakearchibald.com/2017/await-vs-return-vs-return-await/

Reply:
I believe the clean up [after hook](https://github.com/googleapis/nodejs-storage/blob/ea35986a08c6c049865fbdab937c9b2b32cf15af/system-test/storage.ts#L161-L163) logic properly [awaits](https://github.com/googleapis/nodejs-storage/blob/ea35986a08c6c049865fbdab937c9b2b32cf15af/system-test/storage.ts#L3699) for all the buckets to be deleted 👍 .